src/lib/wasm.ts

```
@@ -0,0 +1,66 @@
+import { LuauTemplateResultCode, LuauTemplateResult } from './wasm-types';
+
+// Web worker for wasm execution
+let worker: Worker | null = null;
+
+// Map of pending requests
+const callbacks: Map<number, { resolve: Function; reject: Function }> = new Map();
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)